compiler/ir/ir.tree/src/org/jetbrains/kotlin/ir/util/IrUtils.kt

```diff
@@ -465,46 +470,58 @@ fun irCall(
         ).apply {
             copyTypeAndValueArgumentsFrom(
                 call,
-                dispatchReceiverAsFirstArgument,
-                firstArgumentAsDispatchReceiver
+                receiversAsFirstArguments,
+                firstArgumentsAsReceivers
             )
         }
     }
 
-private fun IrMemberAccessExpression.copyTypeAndValueArgumentsFrom(
+fun IrMemberAccessExpression.copyTypeAndValueArgumentsFrom(
     call: IrMemberAccessExpression,
-    dispatchReceiverAsFirstArgument: Boolean = false,
-    firstArgumentAsDispatchReceiver: Boolean = false
+    receiversAsFirstArguments: Boolean = false,
+    firstArgumentsAsReceivers: Boolean = false
 ) {
     copyTypeArgumentsFrom(call)
 
     var toValueArgumentIndex = 0
     var fromValueArgumentIndex = 0
 
     when {
-        dispatchReceiverAsFirstArgument -> {
+        receiversAsFirstArguments && call.descriptor.dispatchReceiverParameter != null -> {
```

Comment:
Is there any difference between call.dispatchReceiver and call.descriptor.dispatchReceiverParameter in this case? Same question to other usage of descriptors.

Reply:
`call.descriptor.dispatchReceiverParameter` is the parameter in the function definition (i.e., in the function signature) and is null if it's a static function.`call.dispatchReceiver` is the actual argument value (for the parameter) in the call.

The reason I added the descriptor checks is to make sure we only copy the receivers from the "from" call if there are receiver params in the "from" function. Ditto for the other case: Only copy to receivers in the "to" call if there are receiver params in the "to" function.

Reply:
We are getting rid of descriptor usage in lowerings. You can use `call.symbol.owner.dispatchReceiverParameter` for an IR counterpart of `call.descriptor.dispatchReceiverParameter`.

Reply:
@gogabr Done. I had to refactor a little bit because `call` is `IrMemberAccessExpression` which doesn't have the `symbol` property, but `IrFunctionReference` and `IrFunctionAccessExpression` do.